app/models/report.rb

```diff
@@ -4,5 +4,15 @@ class Report < TraineeOnly
 
   belongs_to :course
 
+  delegate :name, to: :course, prefix: true
+  delegate :name, to: :user, prefix: true
+
   validates :date, :today_task, :tmr_task, presence: true
+
+  scope :by_user_id,
+        ->(id){where(user_id: id).order date: :desc, course_id: :asc}
+  scope :by_course_id,
+        ->(ids){where(course_id: ids).order date: :desc, course_id: :asc}
+  scope :by_date,
+        ->(date){where(date: date).order date: :desc, course_id: :asc}
```

Comment:
```suggestion
        ->(date){where(date: date).order(date: :desc, course_id: :asc) if date.present?}
```
Thử thể này xem em, Rồi ở controller không cần if nữa

Reply:
date truyền vào ở đây là params[:filter_date][:filter_date], mà ở controller e check params[:filter_date] vì ko submit filter thì ko có params này ạ. ko submit thì params[:filter_date] là nil, thì lỗi ở params[:filter_date][:filter_date] gọi [] trên nil nên e phải check ở controller :cry: